src/openrct2/drawing/Drawing.String.cpp

```diff
@@ -588,10 +588,26 @@ static void ttf_draw_string_raw_ttf(rct_drawpixelinfo* dpi, const utf8* text, te
                 {
                     if (*src != 0)
                     {
-                        *(dst + 1) = info->palette[3];                   // right
-                        *(dst - 1) = info->palette[3];                   // left
-                        *(dst - width - dstScanSkip) = info->palette[3]; // top
-                        *(dst + width + dstScanSkip) = info->palette[3]; // bottom
+                        // right
+                        if (xx + skipX < dpi->width + dpi->pitch - 1)
```

Comment:
Shouldn't you be able to still compute a start and end for that given rectangle?

Reply:
Sure, this is what this PR does. Keep in mind a rectangle is not a contiguous memory region inside a buffer.

Reply:
We'll I'm gonna trust this fixes it. This could need some refactoring tho, best case a buffer view representing that rectangle providing safe read/writes by offset or x,y coordinates.